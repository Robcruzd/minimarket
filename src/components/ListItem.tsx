import { Dimensions, Image, StyleSheet, View } from "react-native";
import CircleNumber from "./CircleNumber";

const widthCard = Dimensions.get("window").width/4;

const ListItem = (props: {index:number, product:any}) => {
    return (
        <View key={props.index} style={styles.viewList}>
            <CircleNumber quantity={props.product.quantity} 
                style={styles.circleNumber}/>
            <Image
                testID="imageProd"
                style={styles.imageProd}
                source={{
                uri: props.product.urlImage,
                }}
            />
        </View>
    )
}
export default ListItem;

const styles = StyleSheet.create({
    viewList: {
        flexDirection:"row", 
        borderBottomWidth:1, 
        borderBottomColor:"#ddd", 
        alignItems:'center', 
        marginLeft:10
    },
    circleNumber: {
        width:widthCard/2, 
        height:widthCard/2, 
        fontSize:widthCard/4, 
        position: "relative", 
        top:0
    },
    imageProd: {
        width:widthCard, 
        height:widthCard, 
        marginLeft:20
    }
})