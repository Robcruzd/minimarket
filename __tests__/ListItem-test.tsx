import React from 'react';
import { Dimensions, Image } from 'react-native';
import { create } from 'react-test-renderer';
import ListItem from '../src/components/ListItem';

const widthCard = Dimensions.get('window').width / 4;

describe('<ListItem />', () => {
  it('renders correctly', () => {
    const product = {
      quantity: 0,
      urlImage: 'https://example.com/image.png',
    };
    const listItemJson = create(<ListItem index={0} product={product} />).toJSON();
    expect(listItemJson).toMatchSnapshot();
  });

  it('renders the CircleNumber component', () => {
    const product = {
      quantity: 3,
      urlImage: 'https://example.com/image.png',
    };
    const testRenderer = create(<ListItem index={0} product={product} />);
    const circleNumberInstance = testRenderer.root.findByProps({ testID: 'circleNumber' });
    expect(circleNumberInstance).toBeTruthy();
  });

  it('renders the Image component with the correct source', () => {
    const product = {
      quantity: 3,
      urlImage: 'https://example.com/image.png',
    };
    const testRenderer = create(<ListItem index={0} product={product} />);
    const imageInstance = testRenderer.root.findByProps({ testID: 'imageProd' });
    expect(imageInstance.props.source.uri).toBe(product.urlImage);
  });

  it('renders the Image component with the correct style', () => {
    const product = {
      quantity: 3,
      urlImage: 'https://example.com/image.png',
    };
    const testRenderer = create(<ListItem index={0} product={product} />);
    const imageInstance = testRenderer.root.findByProps({ testID: 'imageProd' });
    expect(imageInstance.props.style).toEqual(expect.objectContaining({
      width: widthCard,
      height: widthCard,
      marginLeft: 20,
    }));
  });
});