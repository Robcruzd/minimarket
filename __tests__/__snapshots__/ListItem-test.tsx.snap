// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "borderBottomColor": "#ddd",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "marginLeft": 10,
    }
  }
>
  <Text
    style={
      [
        {
          "backgroundColor": "#A51BB6",
          "borderRadius": 50,
          "color": "#fff",
          "height": 30,
          "left": 5,
          "position": "absolute",
          "textAlign": "center",
          "textAlignVertical": "center",
          "top": 5,
          "width": 30,
          "zIndex": 2,
        },
        {
          "fontSize": 46.875,
          "height": 93.75,
          "position": "relative",
          "top": 0,
          "width": 93.75,
        },
      ]
    }
    testID="circleNumber"
  >
    0
  </Text>
  <Image
    source={
      {
        "uri": "https://example.com/image.png",
      }
    }
    style={
      {
        "height": 187.5,
        "marginLeft": 20,
        "width": 187.5,
      }
    }
    testID="imageProd"
  />
</View>
`;
